docs/types.rst

```diff
@@ -276,6 +276,9 @@ one side.  Examples include ``1.``, ``.1`` and ``1.3``.
 Scientific notation is also supported, where the base can have fractions, while the exponent cannot.
 Examples include ``2e10``, ``-2e10``, ``2e-10``, ``2.5e1``.
 
+Underscores can be used to separate the digits of a numeric literal to aid readability.
+For example, ``123_000``, ``0x2eff_abde``, ``1233_e348_9a`` are all valid. Underscores are only allowed between two digits. For hex literals, underscores are only allowed to separate groups of 4 hex digits.
```

Comment:
I first misread this and thought that there always have to be two digits on each side of the underscore. Maybe ``Literals may not start or end in an underscore.`` (or something better) instead?

Reply:
Hm... my suggestion doesn't exclude ``__`` (two underscores)... not sure - maybe it's fine as it is, maybe you find something better.

Reply:
I am also not yet sure whether I like the idea of having such relaxed grouping rules for decimals. I'll have a look over the text. Thanks.

Reply:
I think the relaxed rules are fine - stricter rules are probably more annoying than helpful I guess.